go/logic/topology_recovery.go

```diff
@@ -1795,9 +1797,15 @@ func executeCheckAndRecoverFunction(analysisEntry inst.ReplicationAnalysis, cand
 	atomic.AddInt64(&countPendingRecoveries, 1)
 	defer atomic.AddInt64(&countPendingRecoveries, -1)
 
+	recoveryDisabledGlobally, recerr := IsRecoveryDisabled()
+	// Check for recovery being disabled globally
+	if recerr != nil {
+		// Unexpected. Shouldn't get this
+		log.Errorf("Unable to determine if recovery is disabled globally: %v", err)
```

Comment:
should we continue execution if `recerr` is not nil?

additionally, we log `err` but error returned from `IsRecoveryDisabled` is named `recerr`. I suggest renaming `recerr` to `err`.

Reply:
Good catch. We should log `recerr`. Keeping it as recerr, just to do as less changes to this function as needed (we use err afterwards)
As for the flow, this is to keep the original one. If we can't determine if recovery is disabled globally or not (which should not happen), we are continuing as it was enabled (`IsRecoveryDisabled` returns `false` in case of error)